### PR TITLE
event endpoint, creation and test

### PR DESCRIPTION
**Pull request by S3JER**
Implements #14

### CODE REVIEW

File: backend/migrations/0001_initial_schema.sql [hunk 1]
```diff
@@ -25,4 +25,11 @@ CREATE TABLE Devices(
...
-)
\ No newline at end of file
+);
+
+CREATE TABLE Events(
+  id INTEGER PRIMARY KEY NOT NULL,
+  task_id INTEGER NOT NULL
+    REFERENCES Tasks(id) ON DELETE CASCADE,
+  start_time DATETIME NOT NULL
+);
```
Comment (mads256h) [inline backend/migrations/0001_initial_schema.sql]: Move to migrations/0002_create_events_table.sql

File: backend/src/main.rs [hunk 2]
```diff
@@ -155,16 +161,61 @@ mod tests {
         uuid.hyphenated().to_string()
     }
 
+    async fn generate_task(
+        app: &mut RouterIntoService<Body>,
+        auth_token: String,
+        device: Device,
```
Comment (mads256h) [inline backend/src/main.rs:167]:
```suggestion
        device: &Device,
```

File: backend/src/main.rs [hunk 3]
```diff
@@ -318,4 +371,41 @@ mod tests {
...
+        if response.status() != StatusCode::OK {
+            let body = response.into_body().collect().await.unwrap().to_bytes();
+            let body = String::from_utf8_lossy(&body);
+            panic!("{}", body);
+        }
+
+        let body = response.into_body().collect().await.unwrap().to_bytes();
+        let response: GetEventsResponse = serde_json::from_slice(&body).unwrap();
+
+        assert_eq!(response.events.first().unwrap().id, event.id);
```
Comment (mads256h) [inline backend/src/main.rs:409]:
```suggestion
        assert_eq!(response.events.first().unwrap(), &event.id);
```

File: backend/src/scheduling/event_creation.rs [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+};
+
+pub async fn _create_event(
+    pool: &SqlitePool,
+    task: Task,
+    start_time: DateTime<Utc>,
+) -> Result<Event, Error> {
+    let event_id = sqlx::query_scalar!(
+        r#"
+        INSERT INTO Events (task_id, start_time) VALUES (?1, ?2)
```
Comment (mads256h) [inline backend/src/scheduling/event_creation.rs:16]:
```suggestion
        INSERT INTO Events (task_id, start_time) VALUES (?, ?)
```

File: backend/src/scheduling/event_creation.rs [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+use sqlx::{Error, SqlitePool};
+
+use crate::data_model::{
+    event::{Event, EventId},
+    task::Task,
+};
+
+pub async fn _create_event(
+    pool: &SqlitePool,
+    task: Task,
```
Comment (mads256h) [inline backend/src/scheduling/event_creation.rs:11]:
```suggestion
    task: &Task,
```

**Review by mads256h — CHANGES_REQUESTED** [review]

**Review by Darkros1245 — APPROVED** [review]

**Review by mads256h — APPROVED** [review]

**Pull request merged**